File: Behavioral/Iterator/CSharp/Song.cs
// Song.cs

public class Song {
  public string Title { get; }
  public string Artist { get; }

  public Song(string title, string artist) {
    Title = title;
    Artist = artist;
  }

  public override string ToString() {
    return $"\"{Title}\" by {Artist}";
  }
}

File: Behavioral/Iterator/Cpp/Song.cpp
// Song.cpp

#include "Song.h"

Song::Song(std::string title, std::string artist) : title(std::move(title)), artist(std::move(artist)) {}

std::ostream& operator<<(std::ostream& os, const Song& song) {
  os << "\"" << song.title << "\" by " << song.artist;
  return os;
}
